File: backend/loja-api/src/main/java/com/accenture/loja/pedido/model/Pedido.java
package com.accenture.loja.pedido.model;

import com.accenture.loja.cliente.model.Cliente;
import com.accenture.loja.shared.enums.StatusPedido;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tb_pedido")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idPedido;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusPedido status;

    private LocalDateTime dataCriacao;
    private LocalDateTime dataReserva;
    private LocalDateTime dataPagamento;
    private LocalDateTime dataCancelamento;

    private String motivoCancelamento;

   @Column(nullable = false)
@Builder.Default
private BigDecimal desconto = BigDecimal.ZERO;

    @Column(nullable = false)
    @Builder.Default
    private BigDecimal totalBruto = BigDecimal.ZERO;

    @Column(nullable = false)
    @Builder.Default
    private BigDecimal totalFinal = BigDecimal.ZERO;

    @OneToMany(mappedBy = "pedido", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ItemPedido> itens = new ArrayList<>();

    public void adicionarItem(ItemPedido item) {
        itens.add(item);
        item.setPedido(this);
    }
}

File: frontend/loja-web/src/types/Pedido.ts
import type { ItemPedido } from './ItemPedido';

export type PedidoStatus = 'CRIADO' | 'RESERVADO' | 'PAGO' | 'CANCELADO';

export interface Pedido {
  idPedido: number;
  clienteId: number;
  status: PedidoStatus;
  dataCriacao: string;        
  dataReserva: string | null;
  desconto: number;           
  totalBruto: number;
  totalFinal: number;
  dataPagamento: string | null;
  dataCancelamento: string | null;
  motivoCancelamento: string | null;
  itens?: ItemPedido[];       
}

export interface CriarPedidoRequest {
  clienteId: number;
  desconto?: number;
  itens: { produtoId: number; quantidade: number }[];
}

export interface CancelarPedidoRequest {
  motivoCancelamento: string;
}